ksqldb-cli/src/test/java/io/confluent/ksql/cli/BasicAuthFunctionalTest.java

```diff
@@ -165,44 +164,55 @@ private static int canMakeCliRequest(final BasicCredentials credentials) {
     )) {
       final RestResponse<ServerInfo> response = restClient.getServerInfo();
       if (response.isSuccessful()) {
-        return Code.OK.getCode();
+        return OK.code();
       }
 
       return response.getErrorMessage().getErrorCode();
     }
   }
 
-  private static Code makeWsRequest(final Optional<BasicCredentials> creds) throws Exception {
-    final WebSocketClient wsClient = new WebSocketClient();
-    wsClient.start();
+  private static int makeWsRequest(final Optional<BasicCredentials> creds) throws Exception {
 
+    Vertx vertx = Vertx.vertx();
+    HttpClient httpClient = null;
     try {
-      final ClientUpgradeRequest request = new ClientUpgradeRequest();
-      creds.ifPresent(c -> {
-        final String authHeader = "Basic " + buildBasicAuthHeader(c.username(), c.password());
-        request.setHeader(HttpHeaders.AUTHORIZATION, authHeader);
-      });
+      httpClient = vertx.createHttpClient();
 
-      final WebSocketListener listener = new WebSocketListener();
-      final URI wsUri = REST_APP.getWsListener().resolve("/ws/query");
+      final URI uri = REST_APP.getWsListener().resolve("/ws/query");
 
-      wsClient.connect(listener, wsUri, request);
+      final MultiMap headers = MultiMap.caseInsensitiveMultiMap();
 
-      assertThat("Response received",
-          listener.latch.await(30, TimeUnit.SECONDS), is(true));
+      creds.ifPresent(c -> {
+        final String authHeader = "Basic " + buildBasicAuthHeader(c.username(), c.password());
+        headers.add(AUTHORIZATION.toString(), authHeader);
+      });
 
-      final Throwable error = listener.error.get();
-      return error == null ? Code.OK : extractStatusCode(error);
+      CompletableFuture<Integer> completableFuture = new CompletableFuture<>();
+
+      httpClient.webSocketAbs(uri.toString(), headers, WebsocketVersion.V07,
```

Comment:
nit: Is there a reason this can't be integrated with `WebsocketUtils#makeWsRequest()`?

Reply:
ack